### ISSUE TITLE
Paper and code

### ISSUE THREAD

**Issue by wnn2000** [M-HuangX/LAION-Beyond#1]
Hi authors,

Congratulations on your paper being accepted by CVPR. I would like to know when your paper and code will be released.

Best,
Nannan

**Comment by M-HuangX** [comment]
Thank you for your attention, I will try my best to upload the codes in next two weeks.

**Comment by wnn2000** [comment]
Hi authors, I have a question after reading your paper.

It’s about your zero-shot method and evaluation. From what I understand, in your setting, you have access to **a set of images from out-of-pretraining (OOP) classes**, so you can fine-tune CLIP on this set, instead of doing a purely zero-shot transfer on just each single image. 

Also, it seems that the vocabulary of the OOP classes is known in advance.

Am I understanding this correctly?

**Comment by M-HuangX** [comment]
So sorry for my late upload of codes and datasets, cuz it's the end of semester recently, and I was quite busy. I will do it as soon as possible. 
For the zero-shot method problem you mentioned, yes, we can access to a set of images from oop classes, so that we can fine-tune CLIP on it.
But accually we don't need to know the vocabulary of oop classes in advance, cuz the procedure of zero-shot method (ZSNL) is, 1. use image features to do clustering and randomly assign a classname (it doesn't matter what kind of classname it will be assigned to a cluster, a natural language word can be proceed by the text encoder is enough. In our codes we use similarity to assign the pseudo label, but we believe the name is not the key for the performance of ZSNL). 2. then these data will be pack just like the dataset used in FSNL to fine tune the CLIP. 3. In the end, use the fine-tuned model to do prediction, we use the pseudo label predicted by the fine-tuned model to find the best match with the ground-truth label so that we can calculate the accuracy.
